### PR TITLE
[SQL] Fix aggregate-related bugs 

### LINKED ISSUE #1578: STDDEV doesn't work for floats or decimals
```sql
CREATE VIEW slot_stats AS SELECT
    STDDEV(x) / AVG(x) AS y
FROM ...
```

where `x` is `DOUBLE` causes. 

```
error[E0277]: the trait bound `dbsp::algebra::F64: num_traits::int::PrimInt` is not satisfied
   --> project018e83eb-b2c0-7c0d-8538-0b749c8fc644/src/main.rs:617:42
    |
617 | ...   Tup15::new(a.0, a.1, sqrt_dN(div_nullNN(minus_dN_dN(a.2.2, div_dN_dN(times_dN_dN(a.2.0, a.2.0), a.2.1)), minus_dN_dN(a.2.1, Some(F6...
    |                                    ^^^^^^^^^^ the trait `num_traits::int::PrimInt` is not implemented for `dbsp::algebra::F64`
    |
    = help: the following other types implement trait `num_traits::int::PrimInt`:
              isize
              i8
              i16
              i32
              i64
              i128
              usize
              u8
            and 4 others
note: required by a bound in `sqllib::operators::div_nullNN`

```

### LINKED ISSUE #1579: `HasZero`, `HasOne` are not implemented for `DECIMAL`
```sql
CREATE VIEW foo AS SELECT
    MAX(x) - MIN(x) AS y,
```

where `x` is `DECIMAL` causes:

```
error[E0277]: the trait bound `rust_decimal::Decimal: HasZero` is not satisfied
   --> project018e83eb-b2c0-7c0d-8538-0b749c8fc644/src/main.rs:614:295
    |
614 | ...).2.1.clone(), indicator::<_, Decimal>(&(*v).4).mul_by_ref(&w).clone()), agg_plus_decimalN_decimalN((*a).2.2.clone(), times_decimalN_d...
    |                                  ^^^^^^^ the trait `HasZero` is not implemented for `rust_decimal::Decimal`
    |
    = help: the following other types implement trait `HasZero`:
              isize
              i8
              i16
              i32
              i64
              i128
              usize
              u8
            and 27 others
```

### LINKED ISSUE #1580: Bug in monotonicity analysis
```
Program status: {'SystemError': 'Exception in thread "main" java.lang.RuntimeException: Expression (Tup4::new((((*t).0).clone()), (((*t).1).clone()), ((*t).2), ((*t).3), ), Tup13::new((((*t).0).clone()), (((*t).1).clone()), ((*t).2), ((*t).3), (((*t).4).clone()), (((*t).5).clone()), ((*t).6), ((*t).7), ((*t).8), ((*t).9), ((*t).10), ((*t).11), (((*t).12).clone()), ), ) 1474 contains multiple instances of t 932 context 1417 (*t) 1418 ((*t).1) 1419 (((*t).1).clone()) 1425 Tup4::new((((*t).0).clone()), (((*t).1).clone()), ((*t).2), ((*t).3), ) 1474 (Tup4::new((((*t).0).clone()), (((*t).1).clone()), ((*t).2), ((*t).3), ), Tup13::new((((*t).0).clone()), (((*t).1).clone()), ((*t).2), ((*t).3), (((*t).4).clone()), (((*t).5).clone()), ((*t).6), ((*t).7), ((*t).8), ((*t).9), ((*t).10), ((*t).11), (((*t).12).clone()), ), ) 
\tat org.dbsp.sqlCompiler.compiler.visitors.inner.RepeatedExpressions.endVisit(RepeatedExpressions.java:78)
\tat org.dbsp.sqlCompiler.compiler.visitors.inner.InnerVisitor.apply(InnerVisitor.java:1069)
\tat org.dbsp.sqlCompiler.compiler.visitors.inner.monotone.MonotoneFunctions.preorder(MonotoneFunctions.java:133)
\tat org.dbsp.sqlCompiler.ir.expression.DBSPClosureExpression.accept(DBSPClosureExpression.java:76)
\tat org.dbsp.sqlCompiler.compiler.visitors.inner.InnerVisitor.apply(InnerVisitor.java:1068)
\tat org.dbsp.sqlCompiler.compiler.visitors.inner.monotone.MonotoneFunctions.apply(MonotoneFunctions.java:299)
\tat org.dbsp.sqlCompiler.compiler.visitors.inner.TranslateVisitor.applyAnalysis(TranslateVisitor.java:89)
\tat org.dbsp.sqlCompiler.compiler.visitors.outer.MonotoneOperators.postorder(MonotoneOperators.java:174)
\tat org.dbsp.sqlCompiler.circuit.operator.DBSPIndexOperator.accept(DBSPIndexOperator.java:59)
\tat org.dbsp.sqlCompiler.circuit.DBSPPartialCircuit.accept(DBSPPartialCircuit.java:125)
\tat org.dbsp.sqlCompiler.circuit.DBSPCircuit.accept(DBSPCircuit.java:65)
\tat org.dbsp.sqlCompiler.compiler.visitors.outer.CircuitVisitor.apply(CircuitVisitor.java:82)
\tat org.dbsp.sqlCompiler.compiler.visitors.outer.CircuitOptimizer$MonotoneAnalyzer.apply(CircuitOptimizer.java:72)
\tat org.dbsp.sqlCompiler.compiler.visitors.outer.CircuitOptimizer$MonotoneAnalyzer.apply(CircuitOptimizer.java:51)
\tat org.dbsp.sqlCompiler.compiler.visitors.outer.Passes.apply(Passes.java:76)
\tat org.dbsp.sqlCompiler.compiler.visitors.outer.Passes.apply(Passes.java:36)
\tat org.dbsp.sqlCompiler.compiler.visitors.outer.CircuitOptimizer.optimize(CircuitOptimizer.java:131)
\tat org.dbsp.sqlCompiler.compiler.DBSPCompiler.optimize(DBSPCompiler.java:323)
\tat org.dbsp.sqlCompiler.CompilerMain.run(CompilerMain.java:165)
\tat org.dbsp.sqlCompiler.CompilerMain.execute(CompilerMain.java:218)
\tat org.dbsp.sqlCompiler.CompilerMain.main(CompilerMain.java:222)


```

### LINKED ISSUE #1581: [SQL] Don't try to simplify multiplications with intervals

### PR DESCRIPTION
**Pull request by mihaibudiu**
Is this a user-visible change (yes/no): no

Fixes #1578 
Fixes #1579 
Fixes #1580 
Fixes #1581

### CODE REVIEW

File: sql-to-dbsp-compiler/lib/sqllib/src/casts.rs [hunk 1]
```diff
@@ -203,21 +203,16 @@ cast_function!(Time, Time, Time, Time);
 #[inline]
 pub fn cast_to_decimal_b(value: bool, precision: u32, scale: u32) -> Decimal {
     let result = if value {
-        Decimal::one()
+        <rust_decimal::Decimal as One>::one()
```
Comment (gz) [inline sql-to-dbsp-compiler/lib/sqllib/src/casts.rs:206]: can't just be?
```suggestion
        <Decimal as One>::one()
```

File: sql-to-dbsp-compiler/lib/sqllib/src/casts.rs [hunk 2]
```diff
@@ -404,23 +399,14 @@ pub fn cast_to_decimalN_fN(value: Option<F32>, precision: u32, scale: u32) -> Op
 
 #[inline]
 pub fn cast_to_decimalN_s(value: String, precision: u32, scale: u32) -> Option<Decimal> {
-    let result = match value.trim().parse() {
-        Err(_) => Some(Decimal::zero()),
-        Ok(x) => Some(x),
-    };
+    let result = Some(value.trim().parse().unwrap());
```
Comment (gz) [inline sql-to-dbsp-compiler/lib/sqllib/src/casts.rs:402]: so this panics (unwrap) instead of returning None?

**Review by gz — APPROVED** [review]

File: sql-to-dbsp-compiler/lib/sqllib/src/casts.rs (hunk 2 above)
Comment (mihaibudiu) [reply sql-to-dbsp-compiler/lib/sqllib/src/casts.rs:402]: that is what our current spec says, and it's consistent with the rest of SQL

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/lib/sqllib/src/casts.rs:402]: so this was in fact a bug (none of the 4 resolved)

File: sql-to-dbsp-compiler/lib/sqllib/src/casts.rs (hunk 1 above)
Comment (mihaibudiu) [reply sql-to-dbsp-compiler/lib/sqllib/src/casts.rs:206]: I am afraid to change it.

**Pull request merged**